js/gamestate.js

```diff
@@ -113,12 +163,10 @@ class GameState {
 
     if (move.op == "beginTurn") {
       this.beginTurn()
-    } else if (move.op == "attack") {
-      this.attack(move.from, move.to)
-    } else if (move.op == "face") {
-      this.face(move.from)
-    } else if (move.op == "play") {
-      this.play(move.from)
+    } else if (move.op == "clickCard") {
+      this.clickCard(move.index, move.containerType)
```

Comment:
somewhere you should document what all the container types are. it seems like having one chunk of comment that just explains all valid moves would be handy.